src/main/java/TaskList.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+import duke.task.Task;
+
+public class TaskList {
+
+    private static ArrayList<Task> taskList;
+
+    public TaskList() {}
+
+    public TaskList(ArrayList<Task> taskList) {
+        TaskList.taskList = taskList;
+    }
+
+    public ArrayList<Task> getTaskList() {
+        return taskList;
+    }
+
+    public void setTaskList(ArrayList<Task> taskList) {
+        taskList = taskList;
+    }
+
+    public void addTask(Task task) {
+        taskList.add(task);
+        printAddedTask(task);
+    }
+
+    public void markAsDone(int taskNo) {
+        Task taskDone = taskList.get(taskNo - 1);
+        taskDone.markAsDone();
+        System.out.println("\tNice! I've marked this task as done:\n\t\t" + taskDone.toString());
+    }
+
+    public void deleteTask(int taskNumber) {
+        Task task = taskList.remove(taskNumber - 1);
+        printDeletedTask(task);
+    }
+
+    private void printAddedTask(Task task) {
```

Comment:
I understand that you did the printing in the TaskList class as you might not want to introduce unnecessary dependencies to other classes. However, I'm not sure it is appropriate for the list to do the printing. I'm unsure myself of how to handle the printing but what do you think of printing classes directly through the command itself, or doing the printing through the Ui (introducing a method that takes in a Command as it's argument)?